cmd/crc/cmd/console.go

```diff
@@ -31,11 +30,9 @@ var consoleCmd = &cobra.Command{
 	Aliases: []string{"dashboard"},
 	Short:   "Open the OpenShift Web Console in the default browser",
 	Long:    `Open the OpenShift Web Console in the default browser or print its URL or credentials`,
-	Run: func(cmd *cobra.Command, args []string) {
-		if renderErr := runConsole(os.Stdout, newMachine(), consolePrintURL, consolePrintCredentials, outputFormat); renderErr != nil {
-			exit.WithMessage(1, renderErr.Error())
-		}
-	},
+	RunE: segmentMiddleware(func(cmd *cobra.Command, args []string) error {
```

Comment:
this can be generalized as an intercept call to allow ANY kind of logging.

So perhaps it is better to use:
```
executeWithLogging(func(cmd *cobra.Command, args []string) error {
```

and make `segmentMiddleware` just an implementation detail of the execution wrapper. Currenrly it feels like this became a dependency, while it is just one of the technologies we use.

Reply:
done